Userland/Libraries/LibGfx/AntiAliasingPainter.cpp

```diff
@@ -587,4 +651,93 @@ void AntiAliasingPainter::fill_rect_with_rounded_corners(IntRect const& a_rect,
+    current_rotated_90deg += mapped_intersection;
+    previous_rotated_90deg += mapped_intersection;
+
+    FloatLine outer_line_current_90 = FloatLine({ current_rotated_90deg, mapped_current_line_b - static_cast<FloatPoint>(mapped_intersection - current_rotated_90deg) });
+    FloatLine outer_line_current_270 = FloatLine({ current_rotated_270deg, mapped_current_line_b - static_cast<FloatPoint>(mapped_intersection - current_rotated_270deg) });
+    FloatLine outer_line_prev_270 = FloatLine({ previous_rotated_270deg, mapped_previous_line_b - static_cast<FloatPoint>(mapped_intersection - previous_rotated_270deg) });
+    FloatLine outer_line_prev_90 = FloatLine({ previous_rotated_90deg, mapped_previous_line_b - static_cast<FloatPoint>(mapped_intersection - previous_rotated_90deg) });
+
+    Optional<FloatPoint> edge_spike_90 = outer_line_current_90.intersected(outer_line_prev_270);
+    Optional<FloatPoint> edge_spike_270 = {};
+
+    if (edge_spike_90.has_value()) {
+        edge_spike_270 = mapped_intersection + (mapped_intersection - edge_spike_90.value());
+    } else {
+        edge_spike_270 = outer_line_current_270.intersected(outer_line_prev_90);
+        if (edge_spike_270.has_value()) {
+            edge_spike_90 = mapped_intersection + (mapped_intersection - edge_spike_270.value());
+        }
+    }
+
+    m_intersection_edge_path.clear();
+    m_intersection_edge_path.move_to(current_rotated_90deg);
+    if (edge_spike_90.has_value())
+        m_intersection_edge_path.line_to(edge_spike_90.value());
+    m_intersection_edge_path.line_to(previous_rotated_270deg);
+
+    m_intersection_edge_path.line_to(current_rotated_270deg);
+    if (edge_spike_270.has_value())
+        m_intersection_edge_path.line_to(edge_spike_270.value());
+    m_intersection_edge_path.line_to(previous_rotated_90deg);
+    m_intersection_edge_path.close();
+    m_underlying_painter.fill_path(m_intersection_edge_path, color);
+}
+
+// rotates a rectangle around 0,0
+FloatQuad AntiAliasingPainter::build_rotated_rectangle(FloatPoint const& direction, float width)
+{
+    double half_size = width / 2;
+    double radian = atan2(direction.y(), direction.x());
+    if (radian < 0) {
+        radian += (2 * M_PI);
+    }
+    // rotated by: (xcosθ−ysinθ ,xsinθ+ycosθ)
+    // p1     p2
+    //
+    //   x,y
+    //
+    // p4     p3
+    double cos_radian = cos(radian);
+    double sin_radian = sin(radian);
+    return FloatQuad(
+        { ((-half_size * cos_radian) - (-half_size * sin_radian)), ((-half_size * sin_radian) + (-half_size * cos_radian)) },
+        { ((half_size * cos_radian) - (-half_size * sin_radian)), ((half_size * sin_radian) + (-half_size * cos_radian)) },
+        { ((half_size * cos_radian)) - (half_size * sin_radian), ((half_size * sin_radian) + (half_size * cos_radian)) },
+        { ((-half_size * cos_radian) - (half_size * sin_radian)), ((-half_size * sin_radian) + (half_size * cos_radian)) });
+}
```

Comment:
Instead of doing all this, try `Gfx::AffineTransform::map_to_quad()` :-)

Reply:
I was testing the rotation with AffineTransform which in deed is much more easier and cleaner, but somehow the rendered image seems suddenly to have more pixel glitches at the edges :-/ So i was leaving the old implemntation as is and added a Todo there.
I tend to say that is a topic for a following fine tune pr. I already found some other issues where i'm curious if i can fix those as well.

Regards